include/roboteam_ai/world/Field.h

```diff
@@ -30,7 +30,6 @@ class Field {
     public:
```

Comment:
I think the Field class in particular needs more refactors than you do now. Many of the functions are functions that should be functions of fieldMessage in my opinion, or simply should not be member functions of field or fieldMessage at all (all the goal vision functions for example are bad, because they do not really have any relevance to Field as a class but are just some function of the field and a worldFrame, and don't really belong here).  I think some consistency in the form of making things functions of either Field or FieldMessage would be nice, as now the 2 abstractions don't really make a lot of sense like this.

Reply:
I indeed agree that the Field class needs to be refactored more. However Jesse and I agreed that the behavior tree needs more attention. So this will be left to do for in the future.